File: lessons/compiled_lang/cs/src/m20.function.cs
using System;

class M20Function
{
    static string[] SortArray(string[] array)
    {
        string[] result = (string[])array.Clone();
        Array.Sort(result);
        return result;
    }

    static void Main()
    {
        string[] array = { "bob", "ed", "steve", "ralph", "joe", "deb", "kate" };
        Console.WriteLine("Current names are: " + string.Join(", ", array));

        string[] result = SortArray(array);
        Console.WriteLine("Sorted names are: " + string.Join(", ", result));
    }
}

File: lessons/compiled_lang/java/src/m20.function.java
import java.util.Arrays;

// Not public - see a00.output.java for why.
class M20Function {
    static String[] sortArray(String[] array) {
        String[] result = array.clone();
        Arrays.sort(result);
        return result;
    }

    public static void main(String[] args) {
        String[] array = {"bob", "ed", "steve", "ralph", "joe", "deb", "kate"};
        System.out.println("Current names are: " + String.join(", ", array));

        String[] result = sortArray(array);
        System.out.println("Sorted names are: " + String.join(", ", result));
    }
}
